src/arrayMethodJoin.js

```diff
@@ -4,8 +4,20 @@
  * Implement method join
  */
 function applyCustomJoin() {
-  [].__proto__.join2 = function(separator) {
-    // write code here
+  [].__proto__.join2 = function(separator = ',') {
+    const indexOfLastElem = this.length - 1;
+    let joinResult = '';
+
+    for (let i = 0; i < this.length; i++) {
+      const element = this[i];
+      const lastElement = (i === indexOfLastElem);
```

Comment:
name `isLastElement` or `isLastIndex` will be more descriptive and more correct in this case
```suggestion
      const isLastIndex = (i === indexOfLastElem);
```